components/director/internal/tenantmapping/cert_service_context_provider.go

```diff
@@ -49,39 +50,73 @@ func (m *certServiceContextProvider) GetObjectContext(ctx context.Context, reqDa
 	// This if is needed to separate the director from ord flow because for the director flow we need to use the internal ID of the subaccount
 	// whereas in the ord flow we expect external IDs in order ord views to work properly(using Automatic Scenario Assignments)
 	log.C(ctx).Infof("Matched component name is %s", matchedComponentName[0])
-	if matchedComponentName[0] == "director" { // Director Flow, do the tenant conversion
-		scopes := "runtime:read runtime:write tenant:read"
 
-		log.C(ctx).Infof("Getting the tenant with external ID: %s", externalTenantID)
-		tenantMapping, err := m.tenantRepo.GetByExternalTenant(ctx, externalTenantID)
-		if err != nil {
-			if apperrors.IsNotFoundError(err) {
-				log.C(ctx).Warningf("Could not find tenant with external ID: %s, error: %s", externalTenantID, err.Error())
+	// the authID in this flow is an OU selected by the Connector
+	externalTenantID := authDetails.AuthID
+
+	if matchedComponentName[0] != "director" { // ORD Flow, set the external tenant ID both for internal and external tenants
```

Comment:
It would be better to check explicitly for "ord" instead of not director. Otherwise if another component is introduced, this logic might not work.